Mathlib/NumberTheory/NumberField/Basic.lean

```diff
@@ -168,6 +168,12 @@ theorem integralBasis_apply (i : Free.ChooseBasisIndex ℤ (𝓞 K)) :
   Basis.localizationLocalization_apply ℚ (nonZeroDivisors ℤ) K (RingOfIntegers.basis K) i
 #align number_field.integral_basis_apply NumberField.integralBasis_apply
 
+theorem mem_span_integralBasis {x : K} :
+    x ∈ Submodule.span ℤ (Set.range (integralBasis K)) ↔ x ∈ 𝓞 K := by
+  rw [integralBasis, Basis.localizationLocalization_span, ← SetLike.mem_coe, ← SetLike.mem_coe,
+    Submodule.map_coe, IsScalarTower.coe_toAlgHom', Submodule.top_coe, Set.image_univ,
+    show Set.range _ = Set.range ((↑) : (𝓞 K) → K) by rfl, Subtype.range_coe]
```

Comment:
Let's split this into two parts: showing the output of `localizationLocalization_span` is actually the algebra, and then connecting this with `integralBasis`:
```suggestion
-- This should be moved, perhaps to Mathlib.Algebra.Algebra.Subalgebra.Tower?
@[simp]
lemma _root_.Subalgebra.range_isScalarTower_toAlgHom {R A} [CommSemiring R] [CommSemiring A]
    [Algebra R A] (S : Subalgebra R A) :
    LinearMap.range (IsScalarTower.toAlgHom R S A) = Subalgebra.toSubmodule S := by
  ext
  simp only [← Submodule.range_subtype (Subalgebra.toSubmodule S), LinearMap.mem_range,
    IsScalarTower.coe_toAlgHom', Subalgebra.mem_toSubmodule]
  rfl

theorem mem_span_integralBasis {x : K} :
    x ∈ Submodule.span ℤ (Set.range (integralBasis K)) ↔ x ∈ 𝓞 K := by
  rw [integralBasis, Basis.localizationLocalization_span, Subalgebra.range_isScalarTower_toAlgHom,
      Subalgebra.mem_toSubmodule]
```